docs/how-to/performance_guidelines.rst

```diff
@@ -275,7 +259,7 @@ allows for more fine-grained control over the execution order of commands,
 which can be beneficial in certain scenarios.
 
 Minimizing memory thrashing
-===========================
+============================
 
 Applications frequently allocating and freeing memory may experience slower
 allocation calls over time. This is expected as memory is released back to the
```

Comment:
Replace 264 to 285 with:
Applications frequently allocating and freeing memory might experience slower allocation calls over time as memory is released back to the operating system. To optimize performance in such scenarios, follow these guidelines:

- Avoid allocating all available memory with ``hipMalloc`` or ``hipHostMalloc``, as this immediately reserves memory and might prevent other applications from using it. This could strain the operating system schedulers or prevent other applications from running on the same GPU.
- Try to allocate memory in suitably sized blocks early in the application's lifecycle and deallocate only when the application doesn't need it anymore. Minimize the number of ``hipMalloc`` and ``hipFree`` calls in your application, particularly in performance-critical areas.
- Consider resorting to other memory types such as ``hipHostMalloc`` or ``hipMallocManaged``, if an application can't allocate sufficient device memory. While the other memory types might not offer similar performance, they allow the application to continue running.
- For supported platforms, use ``hipMallocManaged``, as it allows oversubscription. With the right policies, ``hipMallocManaged`` can maintain most, if not all, ``hipMalloc`` performance. ``hipMallocManaged`` doesn't require an allocation to be resident until it is needed or prefetched, which eases the load on the operating system's schedulers and facilitates multi-tenant scenarios.